core/routes.go

```diff
@@ -26,65 +28,206 @@ func updateRoutes(pg *postgres) {
+			return nil
 		}
 
-		if route.Response.Flightroute.Callsign == "" {
-			existing = append(existing, aircraft)
+		aircrafts = append(aircrafts, aircraft)
+	}
+
+	fmt.Println("Aircrafts that have not have routes processed: ", len(aircrafts))
+	return aircrafts
+}
+
+func checkRouteExists(pg *postgres, aircraftToProcess []Aircraft) (existing []Aircraft, new []Aircraft) {
+
+	var callsignValues []string
+	for _, a := range aircraftToProcess {
+		callsignValues = append(callsignValues, a.Flight)
+	}
+
+	existingRoutes := make(map[string]*Aircraft)
+
+	query := `
+		SELECT id, route_callsign
+		FROM route_data
+		WHERE route_callsign = ANY($1::text[])
+		  AND last_updated IS NOT NULL
+		  AND last_updated > NOW() - INTERVAL '1 hour'`
+
+	rows, err := pg.db.Query(context.Background(), query, callsignValues)
+
+	if err != nil {
+		fmt.Println("checkRouteExists() - Error querying db: ", err)
+		return nil, nil
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		var route Aircraft
+		err := rows.Scan(
+			&route.Id,
+			&route.Flight,
+		)
+
+		if err != nil {
+			fmt.Println("checkRouteExists() - Error scanning rows: ", err)
 			continue
 		}
 
-		routes = append(routes, *route)
-		existing = append(existing, aircraft)
+		existingRoutes[route.Flight] = &route
+	}
 
+	for _, a := range aircraftToProcess {
+		if _, ok := existingRoutes[a.Flight]; ok {
+			existing = append(existing, a)
+		} else {
+			new = append(new, a)
+		}
 	}
 
-	insertRoutes(pg, routes)
+	return existing, new
 
-	MarkProcessed(pg, "route_processed", existing)
+}
+
+func getRoutes(aircrafts []Aircraft) ([]RouteInfo, error) {
+
+	requestBodyData := buildRouteApiRequestBody(aircrafts)
+	requestBodyJson, err := json.Marshal(requestBodyData)
+
+	if err != nil {
+		return nil, err
+	}
+
+	url := "https://adsb.im/api/0/routeset"
+
+	req, err := http.NewRequest("POST", url, bytes.NewReader(requestBodyJson))
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("User-Agent", fmt.Sprintf("Skystats/%s", version))
+	req.Header.Set("Accept", "application/json")
+
+	client := &http.Client{
+		Timeout: 10 * time.Second,
+	}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	var routes []RouteInfo
+	err = json.Unmarshal(body, &routes)
+	if err != nil {
+		return nil, err
+	}
 
+	return routes, nil
 }
 
 func insertRoutes(pg *postgres, routes []RouteInfo) {
 
 	batch := &pgx.Batch{}
-
 	last_updated := time.Now().UTC().Format("2006-01-02 15:04:05-07")
+	countryLookup := CountryIsoToName()
+	queuedCount := 0
 
 	for _, route := range routes {
 
+		// Skip callsigns that were not matched
+		if route.AirportCodesIata == "unknown" {
+			continue
+		}
+
+		// Skip any "unplausible" routes
+		if route.Plausible == false {
+			continue
+		}
+
+		// Skip any multihop routes - for now
```

Comment:
technically that's empty or multihop... (it makes me twitchy when comments and code disagree) 🤷‍♂️